cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1446,6 +1446,12 @@ func findWaitingMachine(ctx context.Context, ds *datastore.RethinkStore, allocat
 // is enabled to clean up networks that were already created.
 func makeNetworks(ctx context.Context, ds *datastore.RethinkStore, ipamer ipam.IPAMer, allocationSpec *machineAllocationSpec, networks allocationNetworkMap, alloc *metal.MachineAllocation) error {
 	for _, n := range networks {
+		if n == nil || n.network == nil {
+			continue
+		}
+		if len(n.network.AddressFamilies) == 0 {
```

Comment:
I don't know if this is possible but can this be dynamically evaluated where the network was retrieved?

Reply:
This simply guards if the network has not AF set